sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,110 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+
+/// Represents ltree specific errors
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum Error {
+   /// LTree labels can only contain [A-Za-z0-9_]
+    #[error("ltree label cotains invalid characters")]
+    InvalidLtreeLabel,
+
+    /// LTree version not supported
+    #[error("ltree version not supported")]
+    InvalidLtreeVersion,
+}
+
+
+/// Represents an postgres ltree. Not that this is an EXTENSION!
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>,
+}
+
+impl PgLTree {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn from_vec(labels: Vec<String>) -> Self {
```

Comment:
True, gotta have checking here. But then it can be a slice of &str

Reply:
I think it's ok to have an unchecked version as long as it's made clear. I'd definitely use it in cases when I'm already encoding the input data to make it be ok for an ltree

Reply:
I'd revert the change to `push`, as it forces an allocation when it could be avoided if the caller already has a `String`. Instead of `from_slice` I'd add a `FromIterator` implementation, and while at it also `IntoIterator` if you'd like

Reply:
Let's have from_slice take an `IntoIterator` of `String` then